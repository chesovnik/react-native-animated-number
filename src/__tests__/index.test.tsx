import * as React from "react";
import { TextInput } from "react-native";
import { create } from "react-test-renderer";
import AnimatedNumber from "../index";

test("It should render with the initial value", () => {
  const component = create(<AnimatedNumber value={100} />);
  const instance = component.root;

  expect(instance.findByType(TextInput).props.value).toEqual("100");

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("It should update to a new value", () => {
  const component = create(<AnimatedNumber value={100} />);
  component.update(<AnimatedNumber value={200} />);

  const input = component.root.findByType(TextInput);
  /** Value should not be changed via props */
  expect(input.props.value).toEqual("100");
});
