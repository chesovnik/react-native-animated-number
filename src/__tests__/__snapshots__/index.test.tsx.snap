// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render with the initial value 1`] = `
<TextInput
  allowFontScaling={true}
  editable={false}
  rejectResponderTermination={true}
  underlineColorAndroid="transparent"
  value="100"
/>
`;
